### PR TITLE
Enhanced shell output for win32 native debugging

### PR DESCRIPTION
**Pull request by yetmorecode**
Tried to improve output of win32 native debugging session with radare2.

Here is an example in current state: http://pastebin.com/FEpLF5JM
This is with the changes: http://pastebin.com/fnMAd5gw

Not quite sure on coding standards or conventions on eprintf()s. Please tell me if something is totally out of shape ;)

### CODE REVIEW

File: libr/debug/p/native/w32.c [hunk 1]
```diff
@@ -151,28 +154,21 @@ static int w32dbg_SeDebugPrivilege() {
...
-static void print_lasterr(const char *str) {
-	/* code from MSDN, :? */
-	LPWSTR pMessage = L"%1!*.*s! %4 %5!*s!";
-	DWORD_PTR pArgs[] = { (DWORD_PTR)4, (DWORD_PTR)2, (DWORD_PTR)L"Bill",  // %1!*.*s!
-		(DWORD_PTR)L"Bob",                                                // %4
-		(DWORD_PTR)6, (DWORD_PTR)L"Bill" };                               // %5!*s!
+static void print_lasterr(const char *caller, char *cause) {
 	WCHAR buffer[200];
 	char cbuffer[100];
-	if (!FormatMessage (FORMAT_MESSAGE_FROM_STRING |
```
Comment (yetmorecode) [inline libr/debug/p/native/w32.c]: This is pretty much just an example copied from MSDN (doesnt even consider 'char *str').

File: libr/debug/p/native/w32.c [hunk 2]
```diff
@@ -151,28 +154,21 @@ static int w32dbg_SeDebugPrivilege() {
...
+				NULL,
+				GetLastError(),
+				LANG_SYSTEM_DEFAULT,
+				&cbuffer,
+				sizeof (cbuffer)-1,
+				NULL)) {
+		eprintf ("Format message failed with 0x%d\n", (ut32)GetLastError ());
 		return;
 	}
-	r_str_wtoc (cbuffer, buffer);
```
Comment (yetmorecode) [inline libr/debug/p/native/w32.c]: FormatMessageA now returns ascii encoded so we dont have to do this any longer.

File: libr/debug/p/native/w32.c [hunk 3]
```diff
@@ -272,20 +268,22 @@ static int w32_first_thread(int pid) {
...
+			if (thid == NULL) {
+        			print_lasterr ((char *)__FUNCTION__, "OpenThread");
                                 goto err_load_th;
+			}
 			CloseHandle (th);
 			return te32.th32ThreadID;
 		}
         } while (Thread32Next (th, &te32));
 err_load_th:    
-        if (ret == -1)
```
Comment (yetmorecode) [inline libr/debug/p/native/w32.c]: ret is always -1 (from initialization above). Does not really indicate an error.

File: libr/debug/p/native/w32.c [hunk 4]
```diff
@@ -318,16 +316,107 @@ static int debug_exception_event (unsigned long code) {
 	return 0;
 }
 
+static char *get_file_name_from_handle(HANDLE handle_file)
```
Comment (yetmorecode) [inline libr/debug/p/native/w32.c]: Reference implementation from MSDN: https://msdn.microsoft.com/en-us/library/windows/desktop/aa366789(v=vs.85).aspx

File: libr/debug/p/native/w32.c [hunk 5]
```diff
@@ -151,28 +154,21 @@ static int w32dbg_SeDebugPrivilege() {
 	return ret;
 }
 
-static void print_lasterr(const char *str) {
-	/* code from MSDN, :? */
-	LPWSTR pMessage = L"%1!*.*s! %4 %5!*s!";
-	DWORD_PTR pArgs[] = { (DWORD_PTR)4, (DWORD_PTR)2, (DWORD_PTR)L"Bill",  // %1!*.*s!
-		(DWORD_PTR)L"Bob",                                                // %4
-		(DWORD_PTR)6, (DWORD_PTR)L"Bill" };                               // %5!*s!
+static void print_lasterr(const char *caller, char *cause) {
```
Comment (ret2libc) [inline libr/debug/p/native/w32.c]: please put a space after print_lasterr

File: libr/debug/p/native/w32.c [hunk 6]
```diff
@@ -272,20 +268,22 @@ static int w32_first_thread(int pid) {
 		eprintf ("w32_thread_list: no thread first\n");
 		return -1;
 	}
-        do {
+
```
Comment (ret2libc) [inline libr/debug/p/native/w32.c]: avoid the whitespaces

File: libr/debug/p/native/w32.c [hunk 7]
```diff
@@ -272,20 +268,22 @@ static int w32_first_thread(int pid) {
...
 	}
-        do {
+        
+	do {
                 /* get all threads of process */
                 if (te32.th32OwnerProcessID == pid) {
 			thid = w32_openthread (THREAD_ALL_ACCESS, 0, te32.th32ThreadID);
-			if (thid == NULL)
+			if (thid == NULL) {
+        			print_lasterr ((char *)__FUNCTION__, "OpenThread");
```
Comment (ret2libc) [inline libr/debug/p/native/w32.c]: white spaces also here

**Comment by ret2libc** [comment]
Hi @yetmorecode, thanks for the PR ;)
Some notes about the style: 
* always use a space after a function name (both in declaration and when you call it)
* remove whitespaces at the end of lines, on empty lines and at the beginning of some lines (git diff will already tell you what's wrong i think)
* braces open on the same line as if/function/while/etc.
* * of  a pointer should be near the variable name

**Comment by radare** [comment]
merged

**Pull request closed without merging**